app/components/animated_number/index.tsx

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useMemo} from 'react';
+import {Animated, Easing, type LayoutChangeEvent, type StyleProp, Text, type TextStyle, View} from 'react-native';
+
+interface Props {
+    animateToNumber: number;
+    fontStyle?: StyleProp<TextStyle>;
+    animationDuration?: number;
+    easing?: ((input: number) => number) | undefined;
+  }
+
+const NUMBERS = Array(10).fill(null).map((_, i) => i);
+
+const usePrevious = (value: number) => {
+    const ref = React.useRef<number|undefined>();
```

Comment:
I think you could avoid the undefined type and the case below by passing in an initial value into `useRef`